### ISSUE TITLE
I am Learning Swift and Curious About a Choice with Elementary 

### ISSUE THREAD

**Issue by maclong9** [sliemeobn/elementary#58]
When creating this library you chose to have options passed with dot-notation inside of the parentheses as opposed to using regular parameters `a(href: "/pages/hello-word")` or as dot-notation passed after the element as a modifier similar to SwiftUI:

``` swift
a {
  "Hello, world
}.href("/pages/hello-world")
```

Not a criticism just wondering why these choices were made so I can better understand the library and language.

**Comment by sliemeobn** [comment]
good question.

"regular" parameters were never an option for me, because there is just a ton of attributes - many shared between multiple elements - and everybody's use case requires different sets of them. choosing general overloads for each tag that work for everyone is impossible.

in general, the main driver was extensibility, because I did not want the library to "pick" the ones that are supported. anyone can easily extend attributes in their packages or "plug-in" packages.

why not "modifier-style": mainly because I found switching from actual HTML to swift code very confusing if the attributes are appended below. so, having them go before the content (just as in HTML) was a clear design goal for me.

I am not against modifier-style APIs at all, but for the "raw basics" of html attributes on tags I found it to be a bad experience.
